src/Core/Silk.NET.Core/Native/SilkMarshal.cs

```diff
@@ -264,6 +314,33 @@ static unsafe string BStrToString(nint ptr)
         public static string MemoryToString(GlobalMemory input, NativeStringEncoding e = NativeStringEncoding.Ansi)
             => PtrToString(input.Handle, e)!; // TODO tolerate a GlobalMemory.Null if we introduce one in the future?
 
+        /// <summary>
+        /// Reads a null-terminated string from the given span, with the given native encoding.
+        /// </summary>
+        /// <param name="input">A span containing a null-terminated string.</param>
+        /// <param name="pin">
+        /// Whether to pin the span such that pointers taken from it are not subject to GC movement.
+        /// </param>
+        /// <param name="e">The encoding of the string in memory.</param>
+        /// <returns>The string read from memory.</returns>
+        public static unsafe string SpanToString
+        (
+            Span<byte> input,
+            bool pin = false,
+            NativeStringEncoding e = NativeStringEncoding.Ansi
+        )
+        {
+            if (pin) // "should" be optimized away if constant
+            {
+                fixed (byte* ptr = input)
+                {
+                    return PtrToString((nint) ptr, e)!;
+                }
+            }
+
+            return PtrToString((nint) Unsafe.AsPointer(ref input.GetPinnableReference()), e)!;
+        }
+        
```

Comment:
- e -> encoding 
```suggestion
        /// <summary>
        /// Reads a null-terminated string from the given span, with the given native encoding.
        /// </summary>
        /// <param name="input">A span containing a null-terminated string.</param>
        /// <param name="pin">
        /// Whether to pin the span such that pointers taken from it are not subject to GC movement.
        /// </param>
        /// <param name="e">The encoding of the string in memory.</param>
        /// <returns>The string read from memory.</returns>
        public static unsafe string SpanToString
        (
            Span<byte> input,
            bool pin = false,
            NativeStringEncoding encoding = NativeStringEncoding.Ansi
        )
        {
            if (pin) // "should" be optimized away if constant
            {
                fixed (byte* ptr = input)
                {
                    return PtrToString((nint) ptr, encoding)!;
                }
            }

            return PtrToString((nint) Unsafe.AsPointer(ref input.GetPinnableReference()), encoding)!;
        }
        
```
- I don't think the pin parameter is needed, and it's suuuper dangerous to default it to false. I'd prefer we get rid of id and always pin, it's a nop on pinned things anyways.
- We need to improve our nullability annotations cleary, nit.

Reply:
> I don't think the pin parameter is needed, and it's suuuper dangerous to default it to false. I'd prefer we get rid of id and always pin, it's a nop on pinned things anyways.

Not on all .NET Standard platforms! Nonetheless, consider it done.